client/state/login/actions.js

```diff
@@ -45,26 +55,49 @@ export const loginUser = ( usernameOrEmail, password ) => dispatch => {
 			password,
 			client_id: config( 'wpcom_signup_id' ),
 			client_secret: config( 'wpcom_signup_key' ),
-		} ).then( ( reponse ) => {
+		} ).then( ( response ) => {
 			dispatch( {
 				type: LOGIN_REQUEST_SUCCESS,
 				usernameOrEmail,
-				data: reponse.body,
+				data: response.body && response.body.data,
 			} );
 		} ).catch( ( error ) => {
-			let errorMessage;
-			const errorKeys = get( error, 'response.body.data.errors' );
-			if ( errorKeys ) {
-				errorMessage = errorKeys.map( errorKey => loginErrorMessages[ errorKey ] ).join( ' ' );
-			} else {
-				errorMessage = get( error, 'response.body.data' ) || error.message;
-			}
+			const errorMessage = getMessageFromHTTPError( error );
+
 			dispatch( {
 				type: LOGIN_REQUEST_FAILURE,
 				usernameOrEmail,
-				error: errorMessage
+				error: errorMessage,
 			} );
 
 			return Promise.reject( errorMessage );
 		} );
 };
+
+/**
+ * Attempt to login a user when a two factor verification code is sent.
+ *
+ * @param  {Number}    two_step_id    Id of the user trying to log in.
+ * @param  {String}    two_step_code  Verification code for the user.
+ * @param  {String}    two_step_nonce Nonce generated for verification code submission.
+ * @param  {Boolean}   remember       Flag for remembering the user for a while after logging in.
+ * @return {Function}                 Action thunk to trigger the login process.
+ */
+export const loginUserWithTwoFactorVerificationCode = ( two_step_id, two_step_code, two_step_nonce, remember ) => {
+	return () => {
+		return request.post( config( 'login_url_xhr' ) )
+			.set( 'Content-Type', 'application/x-www-form-urlencoded' )
+			.accept( 'application/json' )
+			.send( {
+				two_step_id,
+				two_step_code,
+				two_step_nonce,
+				remember,
+				client_id: config( 'wpcom_signup_id' ),
+				client_secret: config( 'wpcom_signup_key' ),
```

Comment:
I think some of those parameters are incorrect. The new login endpoint expects `user_id` and `remember_me`.